From 7db01c6fc1f485acbe8144d0b8ce399f942ba963 Mon Sep 17 00:00:00 2001
From: Laurent FERT <laurent.fert@intel.com>
Date: Wed, 21 Aug 2013 14:11:43 +0200
Subject: [PORT FROM MAIN] Fix broken include

BZ: 132454

Android.mk is including makefiles using LOCAL_PATH uninitialized.
Depending on the previous value of LOCAL_PATH, this may result in
modules being parsed twice and generate build errors.

Make sure LOCAL_PATH is always initialized.
Only include Android.mk from subdirectories when appropriate.

Change-Id: Ibc880070a624839c440f28aa0379f1e0028823ac
Category: aosp improvement
Domain: AOSP-Build
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Laurent FERT <laurent.fert@intel.com>
Reviewed-on: http://android.intel.com:8080/126774
Reviewed-by: Falempe, Jocelyn <jocelyn.falempe@intel.com>
Reviewed-by: Imberton, Guilhem <guilhem.imberton@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 Android.mk |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Android.mk b/Android.mk
index 4712da3..b1c5a05 100644
--- a/Android.mk
+++ b/Android.mk
@@ -23,10 +23,10 @@
 # *not* include it on all devices, so it is safe even with hardware-specific
 # components.
 
-ifneq ($(filter mako occam,$(TARGET_DEVICE)),)
-
 LOCAL_PATH := $(call my-dir)
 
+ifneq ($(filter mako occam,$(TARGET_DEVICE)),)
+
 include $(CLEAR_VARS)
 LOCAL_MODULE       := wpa_supplicant_overlay.conf
 LOCAL_MODULE_TAGS  := optional
@@ -43,6 +43,6 @@ LOCAL_SRC_FILES    := $(LOCAL_MODULE)
 LOCAL_MODULE_PATH  := $(TARGET_OUT_ETC)/wifi
 include $(BUILD_PREBUILT)
 
-endif
-
 include $(call first-makefiles-under,$(LOCAL_PATH))
+
+endif
-- 
1.7.4.1

